shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -100,6 +100,70 @@ - (void)addInternalPlugins;
 - (void)deregisterNotifications;
 @end
 
+@interface KeyboardAnimationView : UIView
+@property(nonatomic) fml::TimePoint startTime;
+@property(nonatomic) NSTimeInterval duration;
+@property(nonatomic) CGFloat from;
+@property(nonatomic) CGFloat to;
+@property(nonatomic) double lastComplete;
+@end
+
+@implementation KeyboardAnimationView : UIView
+@end
+
+@interface KeyboardSpringCurve : NSObject
+@property(nonatomic) double initialVelocity;
+@property(nonatomic) double settlingDuration;
+@property(nonatomic) double response;
+@property(nonatomic) double dampingRatio;
+@property(nonatomic) double omega;
+@end
+
+@implementation KeyboardSpringCurve
+- (id)initWithStiffness:(double)stiffness
+           dampingRatio:(double)dampingRatio
+                   mass:(double)mass
+        initialVelocity:(double)initialVelocity
+       settlingDuration:(double)settlingDuration {
+  self = [super init];
+  if (self) {
+    _dampingRatio = dampingRatio;
+    _initialVelocity = initialVelocity;
+    _settlingDuration = settlingDuration;
+    _response = MAX(1e-5, 2 * M_PI / sqrt(stiffness / mass));
+    _omega = 2 * M_PI / _response;
```

Comment:
Maybe we can delete `response ` property `@property(nonatomic) double response;`

And In init method:
```obj-c
double response = MAX(1e-5, 2 * M_PI / sqrt(stiffness / mass));
_omega = 2 * M_PI / response;
```

Reply:
works for me